src/providers/ldap/sdap_dyndns.c

```diff
@@ -518,65 +521,47 @@ sdap_dyndns_update_recv(struct tevent_req *req)
 struct sdap_dyndns_get_addrs_state {
     struct sdap_id_op* sdap_op;
     struct sss_iface_addr *addresses;
+    const char *network_filter;
 };
 
 static void sdap_dyndns_get_addrs_done(struct tevent_req *subreq);
 static errno_t sdap_dyndns_add_ldap_conn(struct sdap_dyndns_get_addrs_state *state,
                                          struct sdap_handle *sh);
 
 static errno_t get_ifaces_addrs(TALLOC_CTX *mem_ctx,
-                                const char *iface,
+                                const char *iface_filter,
+                                const char *network_filter,
                                 struct sss_iface_addr **_result)
 {
-    struct sss_iface_addr *result_addrs = NULL;
-    struct sss_iface_addr *intf_addrs;
+    struct sss_iface_addr *intf_addrs = NULL;
     TALLOC_CTX *tmp_ctx;
-    char **list_of_intfs;
-    int num_of_intfs;
     errno_t ret;
-    int i;
 
     tmp_ctx = talloc_new(NULL);
     if (tmp_ctx == NULL) {
         ret = ENOMEM;
         goto done;
     }
 
-    ret = split_on_separator(tmp_ctx, iface, ',', true, true, &list_of_intfs,
-                             &num_of_intfs);
+    *_result = NULL;
+    ret = sss_iface_addr_list_get(tmp_ctx, iface_filter, network_filter, &intf_addrs);
     if (ret != EOK) {
-        DEBUG(SSSDBG_MINOR_FAILURE,
-              "Parsing names of interfaces failed - %d:[%s].\n",
-              ret, sss_strerror(ret));
-        goto done;
-    }
-
-    for (i = 0; i < num_of_intfs; i++) {
-        ret = sss_iface_addr_list_get(tmp_ctx, list_of_intfs[i], &intf_addrs);
-        if (ret == EOK) {
-            if (result_addrs != NULL) {
-                /* If there is already an existing list, head of this existing
-                 * list will be considered as parent talloc context for the
-                 * new list.
-                 */
-                talloc_steal(result_addrs, intf_addrs);
-            }
-            sss_iface_addr_concatenate(&result_addrs, intf_addrs);
-        } else if (ret == ENOENT) {
+        if (ret == ENOENT) {
             /* non-critical failure */
             DEBUG(SSSDBG_TRACE_FUNC,
-                  "Cannot get interface %s or there are no addresses "
-                  "bind to it.\n", list_of_intfs[i]);
+                  "Cannot get interface [%s] or there are no addresses "
+                  "bind to it due to filter [%s].\n", iface_filter,
```

Comment:
s/bind/bound/

Reply:
I update the error message to make more sense. The error about interfaces being unreadable is reported elsewhere in the code.